scripts/import-test-connections.js

```diff
@@ -0,0 +1,277 @@
+
+const envs = createTestEnvs([
+  'enterprise',
+  'ldap',
+  'scram',
+  'sharded',
+  'ssh',
+  'tls',
+  'kerberos',
+]);
+
+const connections = [];
+
+if (COMPASS_TEST_ATLAS_URL) {
+  connections.push({
+    favorite: {
+      name: 'Atlas'
+    },
+    connectionOptions: {
+      connectionString: COMPASS_TEST_ATLAS_URL,
+    }
+  })
+}
+
+
+if (COMPASS_TEST_FREE_TIER_URL) {
+  connections.push({
+    favorite: {
+      name: 'Atlas (free tier)'
+    },
+    connectionOptions: {
+      connectionString: COMPASS_TEST_FREE_TIER_URL,
+    }
+  })
+}
+
+if (COMPASS_TEST_SECONDARY_NODE_URL) {
+  connections.push({
+    favorite: {
+      name: 'Atlas (ReadPreference=Secondary)'
+    },
+    connectionOptions: {
+      connectionString: COMPASS_TEST_SECONDARY_NODE_URL,
+    }
+  })
+}
+
+if (COMPASS_TEST_ANALYTICS_NODE_URL) {
+  connections.push({
+    favorite: {
+      name: 'Atlas (nodeType:ANALYTICS)'
+    },
+    connectionOptions: {
+      connectionString: COMPASS_TEST_ANALYTICS_NODE_URL,
+    }
+  })
+}
+
+if (E2E_TESTS_ATLAS_HOST && E2E_TESTS_ATLAS_X509_PEM_PATH) {
+
+  const url = new ConnectionStringUrl(
+    `mongodb+srv://${E2E_TESTS_ATLAS_HOST || ''}/admin`
+  );
+
+  url.searchParams.set('authMechanism', 'MONGODB-X509');
+  url.searchParams.set('tls', 'true');
+  url.searchParams.set('tlsCertificateKeyFile', E2E_TESTS_ATLAS_X509_PEM_PATH);
+  url.searchParams.set('authSource', '$external');
+
+  connections.push({
+    favorite: {
+      name: 'Atlas (X509)'
+    },
+    connectionOptions: {
+      connectionString: url.href,
+    }
+  })
+}
+
+if (COMPASS_TEST_DATA_LAKE_URL) {
+  connections.push({
+    favorite: {
+      name: 'Atlas Data Lake'
+    },
+    connectionOptions: {
+      connectionString: COMPASS_TEST_DATA_LAKE_URL,
+    }
+  })
+}
+
+if (COMPASS_TEST_SERVERLESS_URL) {
+  connections.push({
+    favorite: {
+      name: 'Atlas Serverless'
+    },
+    connectionOptions: {
+      connectionString: COMPASS_TEST_SERVERLESS_URL,
+    }
+  })
+}
+
+[
+  'enterprise',
+  'ldap',
+  'scramReadWriteAnyDatabase',
+  'scramReadWriteAnyDatabaseScramSha1',
+  'scramReadWriteAnyDatabaseScramSha256',
+  'scramOnlyScramSha1',
+  'scramOnlyScramSha256',
+  'scramEncodedPassword',
+  'scramPrivilegesOnNonExistingDatabases',
+  'scramPrivilegesOnNonExistingCollections',
+  'scramAlternateAuthDb',
+  'sharded',
+  'sshPassword',
+  'sshIdentityKey',
+  'sshIdentityKeyWithPassphrase',
+  'tlsUnvalidated',
+  'tlsServerValidation',
+  'tlsServerAndClientValidation',
+  'tlsServerAndClientValidationKeyCrt',
+  'tlsX509',
+  'tlsX509WithSsh',
+  'kerberos',
+  'kerberosAlternate',
+  'kerberosCrossRealm',
+].forEach(envName => connections.push({
+  favorite: {
+    name: envName,
+  },
+  connectionOptions: envs.getConnectionOptions(envName)
+}));
+
+async function main() {
+  const channels = {
+    dev: 'MongoDB Compass Dev',
+    beta: 'MongoDB Compass Beta',
+    stable: 'MongoDB Compass'
+  };
+
+  const channel = process.argv[2];
```

Comment:
Maybe defaulting to `dev` here would make sense? It's the least "destructive" of them all (because doesn't require you to "allow access" multiple times)

Reply:
makes sense to me yeah, I'll also switch to uuid5 so re-running it would replace those instead of duplicating it